src/test/reporters/github.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-ignore
+import milliseconds from 'ms';
+import path from 'path';
+import { BaseReporter, formatFailure, workspaceRelativePath } from './base';
+import {
+  TestCase,
+  FullResult,
+} from '../../../types/testReporter';
+
+type GithubLogType = 'debug' | 'notice' | 'warning' | 'error';
+
+type GithubLogOptions = Partial<{
+  title: string;
+  file: string;
+  col: number;
+  endColumn: number;
+  line: number;
+  endLine: number;
+}>;
+
+class GithubLogger {
+	private _isCI: boolean = process.env.CI === 'true';
+  private _isGithubAction: boolean = process.env.GITHUB_ACTION !== undefined;
```

Comment:
Let's remove these for now. Users can pass --reporter= to play with it.

Reply:
I've removed isCI and shouldLog. 
The `_isGithubAction` is used to format `\n` so left that.